add OpenTofu compatibility

jamesgeddes opened the pull request:
add tofu to AllowedTypes to permit [.tofu files](https://opentofu.org/docs/language/files/)
add tests to ensure that sorting works in .tofu files bsr: fix os.Remove Unhandled error warnings

tests pass

AlexNabokikh commented:
@jamesgeddes thank you very much for your contribution. Please don't hesitate to add `gocognit` and `cyclop` to `linters.disable` in `.golangci.yaml` I don't think they provide much value. `wsl` error has to be addressed, though.

AlexNabokikh commented:
One more thing. Please consider using conventional commit messages, as suggested in the https://github.com/AlexNabokikh/tfsort/blob/master/CONTRIBUTING.md guidelines.

jamesgeddes commented:
Apologies for missing the conventional commit, I have now fixed. Have also reverted to your original tests. Tests passing, so will be happy to make any futher amendments to pass CI. Thanks!
